pkg/controller/clustersync/clustersync_controller_test.go

```diff
@@ -0,0 +1,1917 @@
+package clustersync
```

Comment:
👍  I really liked how you organized the test cases, testing one aspect of the reconciler at a time. It may be useful to update the clusterdeployment controller tests to something like this in the future.